packages/react/src/components/Grid/Grid.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * Copyright IBM Corp. 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { settings } from 'carbon-components';
+import { breakpoints } from '@carbon/layout';
+
+import { range } from '../../tools/array';
+
+const { prefix } = settings;
+
+// getter functions needed due to error in storybook
+// addons knobs that requires a copy of this object to avoid
+const getValidColWidths = () => ({
+  sm: [false, true, 'auto', ...range({ end: breakpoints.sm.columns })],
+  md: [false, true, 'auto', ...range({ end: breakpoints.md.columns })],
+  lgPlus: [false, true, 'auto', ...range({ end: breakpoints.lg.columns })],
+});
+
+const getValidColOffsets = () => ({
+  sm: range({ end: breakpoints.sm.columns - 1 }),
+  md: range({ end: breakpoints.md.columns - 1 }),
+  lgPlus: range({ end: breakpoints.lg.columns - 1 }),
+});
+
+const VALID_COL_WIDTHS = getValidColWidths();
+const VALID_COL_OFFSETS = getValidColOffsets();
+
+export const Grid = ({
+  as = 'div',
+  condensed = false,
+  fullWidth = false,
+  noGutter = false,
+  className = '',
+  children = null,
+  ...rest
+}) =>
+  React.createElement(
+    as,
+    {
+      ...rest,
+      className: classNames(
+        `${prefix}--grid`,
+        condensed && `${prefix}--grid--condensed`,
+        fullWidth && `${prefix}--grid--full-width`,
+        noGutter === true && `${prefix}--no-gutter`, // `true` === both
+        noGutter !== true && noGutter && `${prefix}--no-gutter--${noGutter}`, // `left` || `right`
+        className
+      ),
+    },
+    children
+  );
+
+Grid.propTypes = {
+  /**
+   * Provide a custom element to render instead of the default <div>
+   */
+  as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
```

Comment:
Maybe `elementType` might be helpful here? Always forget which one is for elements lol

```suggestion
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
```

Reply:
e668a2